daliuge-engine/dlg/drop.py

```diff
@@ -427,28 +427,26 @@ def get_param_value(attr_name, default_value):
             has_app_param = (
                     "applicationArgs" in kwargs and attr_name in kwargs["applicationArgs"]
             )
-            param = default_value
-            if has_component_param and has_app_param:
-                logger.warning(
-                    "Drop has both component and app param %s. Using component param.",
-                    attr_name
-                )
+            # c_param = a_param = default_value
             if has_component_param:
-                param = kwargs.get(attr_name)
+                c_param = kwargs.get(attr_name)
             elif has_app_param:
-                if kwargs["applicationArgs"].get(attr_name).usage in [
+                if kwargs["applicationArgs"].get(attr_name).usage not in [
                     "InputPort",
                     "OutputPort",
                     "InputOutput",
                 ]:
-                    # inp = kwargs["input"]
-                    # param = pickle.loads(
-                    #     droputils.allDropContents()
-                    #     )
-                    pass
-                else:
-                    param = kwargs["applicationArgs"].get(attr_name).value
-            return param
+                    a_param = kwargs["applicationArgs"].get(attr_name).value
+            if has_component_param:
+                # logger.warning(
+                #     "Drop has both component and app param %s. Using component param.",
+                #     attr_name
+                # )
+                return c_param
+            elif has_app_param:
+                return a_param
+            else:
+                return default_value
```

Comment:
Aren't we having some scope creep between our definitions of component vs attribute here?

Reply:
that is legacy and the commented part was there before. The issue is that in the Apps very often the parameters are set using dlg_*_param functions directly into class attributes and those seem to end up as component_parameters (I don't quite know how, though). Might be worthwhile to have another look and make sure they are put in the correct category. I found it always quite annoying that we always had these log-messages about duplicate parameters.